features/305-microservice/microservice.md

```diff
@@ -0,0 +1,85 @@
+# Microservice
+
+## Overview
+
+The document specifies the details of `type: http-service` of a deployment item.
+
+## Configuration of the deployment item
+
+```yaml
+deployments:
+  my-awesome-service:
+    type: http-service
```

Comment:
how would the config looks like if users didn't pass a docker file, but instead, point to a path with node.js looks like? Is the current configuration compatible with that? Or we need a new `type` (prefer no)?

So other config such as `context` need to be considered.

Reply:
The type `http-service` is docker only. In the future we can have other types like `nodejs-service` which itself is just an abstraction over `http-service`. That is we can provide the Dockerfile for them given that the project follows some assumed conventions.

The Dockerfile may look like

```dockerfile
FROM node:10-alpine
WORKDIR /usr/src/app
COPY package.json package-lock.json .
RUN npm ci
COPY . .
RUN npm run prepare
EXPOSE 8080
ENTRYPOINT ["npm", "start"]
```

So the developer is expected to define `prepare` and `start` in their package.json. `prepare` is supposed to build their code (because they may be using TypeScript). `start`, as the name suggests, start the server.